### PR TITLE
Logging and Exception Management assignment

### PR DESCRIPTION
**Pull request by Sartaj-ti**

### CODE REVIEW

File: fast_api_als/services/enrich_lead.py [hunk 1]
```diff
@@ -12,4 +13,7 @@
 
 
 def get_enriched_lead_json(adf_json: dict) -> dict:
-    pass
\ No newline at end of file
+    adf_obj = json.loads(adf_json, object_hook=lambda d: SimpleNamespace(**d))
+    if not adf_obj:
+        raise TypeError(f'{adf_json} cannot be converted to object')
```
Comment (naman-TI) [inline fast_api_als/services/enrich_lead.py:19]: check for KeyError as well

File: fast_api_als/routers/submit_lead.py [hunk 2]
```diff
@@ -38,7 +38,7 @@ async def submit(file: Request, apikey: APIKey = Depends(get_api_key)):
     
     if not db_helper_session.verify_api_key(apikey):
         # throw proper fastpi.HTTPException
-        pass
+        raise fastapi.HTTPException(status_code=404, detail='Wrong API key')
```
Comment (naman-TI) [inline fast_api_als/routers/submit_lead.py]: should it be 400?

File: fast_api_als/routers/submit_lead.py [hunk 3]
```diff
@@ -197,6 +197,8 @@ async def submit(file: Request, apikey: APIKey = Depends(get_api_key)):
             }
         }
         res = sqs_helper_session.send_message(message)
```
Comment (naman-TI) [inline fast_api_als/routers/submit_lead.py]: always need to handle exceptions when using third party APIs

Comment (naman-TI) [reply fast_api_als/routers/submit_lead.py]: line 215 also and all instances where AWS apis are being used

**Review by naman-TI — CHANGES_REQUESTED** [review]
This is not approved. Need to make changes to 3 more files and concrete logging/exception handling is not present in the 5 files changed.

**Comment by Sartaj-ti** [comment]
Okk sir, I will make the necessary changes and submit again.

File: fast_api_als/utils/adf.py [hunk 4]
```diff
@@ -39,17 +54,24 @@ def is_nan(x):
...
-    names = input_json['customer']['contact']['name']
-    make = input_json['vehicle']['make']
+    try:
+        input_json = input_json['adf']['prospect']
+        zipcode = input_json['customer']['contact']['address']['postalcode']
+        email = input_json['customer']['contact'].get('email', None)
+        phone = input_json['customer']['contact'].get('phone', None)
+        names = input_json['customer']['contact']['name']
+        make = input_json['vehicle']['make']
+    except Exception as e:
```
Comment (naman-TI) [inline fast_api_als/utils/adf.py:72]: if this is KeyError, why do you need to catch the generic 'Exception'

**Pull request closed without merging**